pnpm schema fails when the monorepo root is named skeleton-*

### Current Behavior

In generate-scemas.ts, the code matches the directory by using 
```js
const framework = path.match(/skeleton-([^/]+)/)?.[1];
```
If the root of monorepo is named skeleton-*, for example, skeleton-ui, the schema will be generated under the directory of ui folder instead of the supposedly svelte and react folder

For example if the directory is
```
/home/louis/repos/skeleton-ui-dev/skeleton/sites/next.skeleton.dev
```
then the folder will be generated as
```
/schema/ui-dev
```

### Expected Behavior

schema should be generated under svelte and react folder



### Steps To Reproduce

```
git clone -b next https://github.com/skeletonlabs/skeleton.git skeleton-ui && cd skeleton-ui

then run pnpm i and schema
```

### Link to Reproduction / Stackblitz

_No response_

### More Information

_No response_

I talked with @endigo9740  regarding another error that might be caused by the path issue; I have documented it here.

## Issue Description

After running the new schema, the following error occurred:

```
dev:astro: [vite:css] [postcss] Cannot find module '/home/louis/repos/skeleton-ui-dev/skeleton/sites/next.skeleton.dev/node_modules/@skeletonlabs/skeleton/dist/themes/index.cjs'
```

The file `themes/index.cjs` could not be found. The issue might be related to directory structure or dependency installation.


- The issue was resolved by switching to a directory without "skeleton-something" in the path, so I suspect it might be a directory error.

I don't think that this is a significant issue since contribution guidelines can be updated to mention these issues.



OS Tested:
Ubuntu 22.04 jammy(on the Windows Subsystem for Linux)
Ubuntu 24.10 Oracular Oriole

--- a/sites/next.skeleton.dev/scripts/generate-schemas.ts
+++ b/sites/next.skeleton.dev/scripts/generate-schemas.ts
@@ -15,7 +15,7 @@ function toKebabCase(str: string) {
 
 async function processFile(path: string): Promise<number> {
 	const component = basename(dirname(path));
-	const framework = path.match(/skeleton-([^/]+)/)?.[1];
+	const framework = path.match(/skeleton-([^/]+)(?=\/src\/lib\/components)/)?.[1];
 	if (!framework) {
 		throw new Error(`Invalid framework path: ${path}`);
 	}

--- a/sites/next.skeleton.dev/scripts/generate-schemas.ts
+++ b/sites/next.skeleton.dev/scripts/generate-schemas.ts
@@ -15,7 +15,9 @@ function toKebabCase(str: string) {
 
 async function processFile(path: string): Promise<number> {
 	const component = basename(dirname(path));
-	const framework = path.match(/skeleton-([^/]+)(?=\/src\/lib\/components)/)?.[1];
+	const matches = path.match(/skeleton-([^/]+)(?=\/|$)/g);
+	const match = matches?.at(-1);
+	const framework = match?.replace('skeleton-', '');
 	if (!framework) {
 		throw new Error(`Invalid framework path: ${path}`);
 	}

